deployments/fuji/usdc/deploy.ts

```diff
@@ -1,171 +1,123 @@
-  );
-
-  let usdcImplementation = await deploymentManager.clone(
-    cloneAddr.usdcImplementation,
-    [],
-    cloneNetwork
-  );
-
-  let usdc;
-  let usdcProxy = await deploymentManager.clone(
-    cloneAddr.usdcProxy,
-    [usdcImplementation.address],
-    cloneNetwork
-  );
+async function deployContracts(deploymentManager: DeploymentManager, deploySpec: DeploySpec): Promise<Deployed> {
+  const { ethers } = deploymentManager.hre;
+  const signer = await deploymentManager.getSigner();
 
-  debug(`Changing admin of USDC proxy to ${usdcProxyAdmin.address}`);
-  await deploymentManager.asyncCallWithRetry(
-    (signer_) => wait(usdcProxy.connect(signer_).changeAdmin(usdcProxyAdmin.address))
-  )
-  usdc = usdcImplementation.attach(usdcProxy.address);
-  // Give signer 10,000 USDC
-  debug(`Initializing USDC`);
-  await deploymentManager.asyncCallWithRetry(
-    (signer_) => wait(
-      usdc.connect(signer_).initialize(
-        'USD Coin',
-        'USDC',
-        'USD',
-        6,
-        signerAddress,
-        signerAddress,
-        signerAddress,
-        signerAddress
-      )
-    )
+  // Deploy fauceteer (XXX probably first in order to mint cloned gov COMP to it)
+  const fauceteer = await deploymentManager.deploy('fauceteer', 'test/Fauceteer.sol', []);
+
+  // XXX clone
+  const governor = await deploymentManager.deploy('governor', 'test/GovernorSimple.sol', []);
+  const timelock = await deploymentManager.deploy('timelock', 'test/SimpleTimelock.sol', [governor.address]);
+
+  // XXX will fail if gov already has a diff timelock, and technically should otherwise ensure admin
+  //  but we are anyway replacing gov simple
+  await deploymentManager.idempotent(
+    async () => !sameAddress(await governor.timelock(), timelock.address),
+    async () => {
+      debug(`Initializing GovSimple`);
+      await wait(governor.initialize(timelock.address, [signer.address]));
+    }
   );
 
-  let wbtc = await deploymentManager.clone(cloneAddr.wbtc, [], cloneNetwork);
-
-  let wavax = await deploymentManager.clone(cloneAddr.wavax, [], cloneNetwork);
-  // Give admin 0.01 WAVAX tokens [this is a precious resource here!]
-  debug(`Minting some WAVAX`);
-  await deploymentManager.asyncCallWithRetry(
-    (signer_) => wait(wavax.connect(signer_).deposit({ value: exp(0.01, 18) }))
+  const usdcProxyAdmin = await deploymentManager.deploy('USDC:admin', 'vendor/proxy/transparent/ProxyAdmin.sol', []);
+  const usdcImpl = await deploymentManager.clone('USDC:implementation', cloneAddr.usdcImpl, [], cloneNetwork);
+  const usdcProxy = await deploymentManager.clone('USDC', cloneAddr.usdcProxy, [usdcImpl.address], cloneNetwork);
+  const usdcProxyAdminSlot = '0x10d6a54a4754c8869d6886b5f5d7fbfa5b4522237ea5c60d11bc4e7a1ff9390b';
+  const USDC = usdcImpl.attach(usdcProxy.address);
+
+  await deploymentManager.idempotent(
+    async () => !sameAddress(await ethers.provider.getStorageAt(usdcProxy.address, usdcProxyAdminSlot), usdcProxyAdmin.address),
```

Comment:
Just that it adds the retry on the action and awaits the condition, and theoretically we might to add something like a label or move the debug message to a third arg so it would always say what was done/not done without explicit printing in the actions

Reply:
The idempotent in the branch name more refers to the additional alias arg you saw in `deploy`, `clone`, and `existing`